### ISSUE TITLE
Old version of mod deleted even if new version failed to download

### ISSUE THREAD

**Issue by Kenqr** [IllusionMods/KKManager#92]
When updating existing mods to a new version, if the new version failed to download, the old version will still be deleted.
This leads to missing mods until the new version can be downloaded.
Currently I have a 900mb map mod that takes about 10 hours to download, and the download has already failed many times, while the old version is gone.
Can you add a check before deleting old version of mods?

**Comment by ManlyMarco** [comment]
Yeah, it's a known issue, this is because KKM doesn't know the GUID and so can't reliably tell if it's the same mod when the file name is changed (like when version number is in the name). Removing first is done to avoid an issue where user aborts the download and ends up with multiple versions of a zipmod or incompatible zipmods, which then breaks the game (this happened many times before this was implemented).

This will be fixed whenever the server side gets changed to a more dedicated solution that can supply necessary metadata.

**Comment by Kenqr** [comment]
Since most version numbers in mod names follow a few patterns, KKM should be able to detect most mod updates correctly.
As for other tasks, they can be done in the same order as now.

Download Mod A v1.1
Delete Mod A v1.0
Download Mod B v2
Delete Mod B v1
Delete Mod C
Delete Mod D
Download Mod E
Download Mod F

There should be no downside if some mod updates are not detected correctly.

**Comment by sjcodan** [comment]
Program just deleted hundreds of my mods. 
If its gonna be this volatile, put a goddamn "Backup your mod folder" warning on it.

**Comment by ManlyMarco** [comment]
@sjcodan Was it because of this issue or are you using an old KKManager version and finished downloads got lost? If the latter, update KKManager.
